controllers/mirrorpeer_controller.go

```diff
@@ -224,6 +281,63 @@ func (r *MirrorPeerReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	return ctrl.Result{Requeue: true}, nil
 }
 
+func (r *MirrorPeerReconciler) deleteResources(mirrorPeer multiclusterv1alpha1.MirrorPeer) error {
+	logger := log.FromContext(context.TODO())
+
+	for i := range mirrorPeer.Spec.Items {
+		managedClusterAddOn := addonapiv1alpha1.ManagedClusterAddOn{
```

Comment:
multiple mirrorpeers could be using the same addon. we can not delete it when only one of them is deleted.
We should probably set ownerref and let garbage collection handle rest.